arc_cache.go

```diff
@@ -17,7 +18,9 @@ type cacheSize int
 // size. This provides block access-time improvements, allowing
 // to short-cut many searches without querying the underlying datastore.
 type arccache struct {
-	cache      *lru.TwoQueueCache
+	cache *lru.TwoQueueCache
+	lks   [256]sync.Mutex
```

Comment:
Thinking through this some more, we can actually use `sync.RWMutex` and only take the _read_ lock on `Get`, `Has`, and `GetSize`. That should reduce contention on read.